### ISSUE TITLE
Crashlytics ANR 

### ISSUE THREAD

**Issue by ExpensiveBelly** [mParticle/mparticle-android-sdk#222]
```
Caused by com.github.anrwatchdog.ANRError$$$_Thread: main (state = TIMED_WAITING)
       at sun.misc.Unsafe.park(Unsafe.java)
       at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:230)
       at kotlinx.coroutines.BlockingCoroutine.joinBlocking(BlockingCoroutine.java:90)
       at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(BuildersKt__BuildersKt.java:61)
       at kotlinx.coroutines.BuildersKt.runBlocking(BuildersKt.java:1)
       at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(BuildersKt__BuildersKt.java:40)
       at kotlinx.coroutines.BuildersKt.runBlocking$default(BuildersKt.java:1)
       at bo.app.e.a(SourceFile:4)
       at bo.app.e.getBoolean(SourceFile:1)
       at bo.app.v4.a(SourceFile:1)
       at com.appboy.Appboy.isDisabled(SourceFile:10)
       at com.appboy.Appboy.registerPushToken(SourceFile:1)
       at com.appboy.Appboy.registerAppboyPushMessages(SourceFile:1)
       at com.mparticle.kits.AppboyKit.onPushRegistration(AppboyKit.java:439)
       at com.mparticle.kits.KitManagerImpl.initializeKit(KitManagerImpl.java:302)
       at com.mparticle.kits.KitManagerImpl.configureKits(KitManagerImpl.java:226)
       at com.mparticle.kits.KitManagerImpl.lambda$updateKits$0(KitManagerImpl.java:157)
       at com.mparticle.kits.KitManagerImpl.$r8$lambda$lHLpvmq04PtgoJLPu-eSVCfcdgM(KitManagerImpl.java)
       at com.appboy.Appboy$$InternalSyntheticLambda$0$4448c33babd421c435788a97f10540b70d9ffc5749c14496b377d14d5cd989bd$0.run$bridge(Appboy.java:22)
       at android.os.Handler.handleCallback(Handler.java:883)
       at android.os.Handler.dispatchMessage(Handler.java:100)
       at android.os.Looper.loop(Looper.java:237)
       at android.app.ActivityThread.main(ActivityThread.java:8167)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100)
```

Just to let you know, we have this in our Crashlytics logs in v5.38.2

**Comment by mchuangatmp** [comment]
@ExpensiveBelly thanks for the feedback, we have added many fixes since v5.38.2...can you update to latest v5.44.1 (core and appboy kit) to see if that fixes things?

**Comment by ExpensiveBelly** [comment]
I'll close for now since I haven't seen it again, I'll re-open if need be. Thanks
